### ISSUE TITLE
Mild LEST_LOST_EXCEPTION_STACK_TRACE false positive

### ISSUE THREAD

**Issue by io7m** [mebigfatguy/fb-contrib#290]
Hello!

This one's arguably "correct"  but common use cases might mark this one as a false positive.

Essentially:

```
try {
  f();
} catch (UncheckedIOException e) {
  throw e.getCause();
}
```

The method `f` is an arbitrary method that throws an `UncheckedIOException`, possibly because it uses lambda/stream operations internally that can't throw checked exceptions. The `UncheckedIOException` type is ostensibly intended to wrap an `IOException` and to be discarded at the first opportunity. However, the above code will trigger a `LEST_LOST_EXCEPTION_STACK_TRACE` bug.

**Comment by ThrawnCA** [comment]
You're losing a step in the stack trace, and it's impossible to say how large a step - how far down did the code convert checked to unchecked? FindBugs can't easily tell - so the detector is correct in principle.

If you really want to do this and are sure that you don't need the missing stack trace information, then this is a good candidate for suppressing the warning on this method.

**Comment by mebigfatguy** [comment]
Thanks for the report! Given the limited use case for this exception, and it's likely use to escape lambdas, i tend to agree that special handling in this case is fine. We already do is for InvocationTargetException

**Comment by mebigfatguy** [comment]
Should be fixed on HEAD for 7.4.2 release
